compiler/semtypes.nim

```diff
@@ -772,6 +772,8 @@ proc semRecordCase(c: PContext, n: PNode, check: var IntSet, pos: var int,
                  formatMissingEnums(c, a))
     else:
       localError(c.config, a.info, "not all cases are covered")
+  if n.len == 2 and chckCovered:
+    incl(a[0].sym.flags, sfDiscriminantAllInOne)
```

Comment:
I don't understand this flag nor its implications.

Reply:
It means users `of` branchs covered all case, no need inject `FieldDiscriminantCheck` code